Feature/issue 90 process case

ciozi137 opened the pull request:
### Summary
This PR addresses Issue #90 by making process name matching case-insensitive to align with LabVIEW's behavior when recognizing overrides.

### Problem
LabVIEW recognizes overrides regardless of case (e.g., both `Process.vi` and `process.vi` are valid), but our current implementation was case-sensitive. This mismatch could cause issues when process names didn't exactly match the expected case.

### Changes Made

- `GetCallingProcessCloneName.vi`: Updated to perform case-insensitive matching for `Process.vi` names
- `WaitOnProcessSync.vi`: Modified to use `GetCallingProcessCloneName.vi` for consistent process name resolution, ensuring the same matching algorithm is applied during both registration and synchronization

<img width="631" height="335" alt="image" src="https://github.com/user-attachments/assets/16798093-6fb2-4216-84d0-e710b09e71b6" />

<img width="967" height="545" alt="image" src="https://github.com/user-attachments/assets/299e7a82-0ac8-411c-8b68-5c9e81f5ed93" />

francois-normandin committed:
Merge tag '1.4.0' into develop

ciozi137 committed:
Issue 90:
- Change GetCallingProcessCloneName.vi to be agnostic to Process.vi case (match on Process.vi or process.vi)
- Change WaitOnProcessSync.vi to call GetCallingProcessCloneName.vi to ensure same match algorithm as was used to register the process

ciozi137 committed:
Add unit tests for Issue #90

Tom-McQuillan commented:
In principle I am supportive of this.
Although, I am curious why P/process.vi would be spelt differently.
Is it common to create the process vi without using the template tool?

ciozi137 commented:
@Tom-McQuillan good question. I'm not sure why I ran into this problem. Maybe I had created 'process.vi' and dropped in the template:
<img width="206" height="182" alt="image" src="https://github.com/user-attachments/assets/98dbcff9-fb46-4216-90ec-e15c116573e1" />

Maybe I temporarily renamed it 'DoNotRunProcess.vi' for some other debugging purpose and then inadvertently renamed it 'process.vi' when I switched back.

The main point is that LabVIEW doesn't recognize a difference and wouldn't, didn't, alert the user if the case didn't match. So hoping to save someone else in the future.

I don't know if there is a more elegant fix available instead of:
<img width="631" height="335" alt="481616869-16798093-6fb2-4216-84d0-e710b09e71b6" src="https://github.com/user-attachments/assets/6d4649e6-2e6b-473f-895f-e0e47ddfb33f" />
